sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedPartitionReader.scala

```diff
@@ -72,12 +74,20 @@ private case class ChangeFeedPartitionReader
 
   private val rowSerializer: ExpressionEncoder.Serializer[Row] = RowSerializerPool.getOrCreateSerializer(readSchema)
 
-  private lazy val iterator: PeekingIterator[ObjectNode] =  Iterators.peekingIterator(
-    new CosmosPagedIterable[ObjectNode](
-      cosmosAsyncContainer.queryChangeFeed(changeFeedRequestOptions, classOf[ObjectNode]),
+  private lazy val iterator: PeekingIterator[ObjectNode] = Iterators.peekingIterator(
+    new TransientIOErrorsRetryingIterator(
```

Comment:
Is this related to query plan or the retry for query paging?

Reply:
Spark specific Retry policy for connectivity issues (also client-side generated) for query and changefeed

Reply:
Instead of allowing to change the GoneAndRetryPolicy in the SDK - and instead triggering retries in Spark layer

Reply:
Should this be added on the Description just as FYI?

Reply:
Yes makes sense - will add a comment explaining this.